pkg/controller/utils/utils.go

```diff
@@ -438,32 +439,33 @@ func WaitToAddResourceWatch(controller controller.Controller, client kubernetes.
 				duration = maxDuration
 			}
 			ticker.Reset(duration)
-			if isResourceReady(client, obj.GetObjectKind().GroupVersionKind().Kind) {
-				err := controller.Watch(&source.Kind{Type: obj}, &handler.EnqueueRequestForObject{})
-				if err != nil {
-					log.Info("failed to watch %s resource: %v. Will retry to add watch", obj.GetObjectKind().GroupVersionKind().Kind, err)
-				} else {
-					flag.MarkAsReady()
-					return
-				}
+			if ok, err := isResourceReady(client, obj.GetObjectKind().GroupVersionKind().Kind); err != nil {
+				log.Info("failed to list registered groups and resources - will retry: %v", err)
+			} else if !ok {
+				log.Info("waiting for %s resource to be registered - will retry", obj.GetObjectKind().GroupVersionKind().Kind)
```

Comment:
if we add the Kind to a context logger we can remove the string formatting from this one and the one below.